cocotb/binary.py

```diff
@@ -32,27 +32,59 @@
 
 import os
 import random
+import re
 import warnings
 
+
+RESOLVE_TO_0     = "-lL"
+RESOLVE_TO_1     = "hH"
+RESOLVE_TO_ERROR = "xXzZuUwW"  # Resolve to a ValueError() since these usually mean something is wrong
```

Comment:
What's the motivation for moving these to global public constants?

Reply:
I moved them so that they were defined before both `Resolver` and `BinaryValue` classes.
`BinaryValue` calls `resolver.resolve()` so it was a chicken/egg problem.

Reply:
It's not a chicken and egg problem though - `resolver.resolve` can use `BinaryValue._resolve_to_1` without any issue.

If you think they're clearer at the top of the file, then please rename them to start with an underscore (ie, to be private), and remove the `BinaryValue` attributes completely.